runtime/registry.go

```diff
@@ -20,7 +20,7 @@ func (r *Runtime) CreateInstance(ctx context.Context, inst *drivers.Instance) er
 	// Check OLAP connection
 	olap, err := drivers.Open(inst.OLAPDriver, inst.OLAPDSN, r.logger)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to open database, Rill might already be running. DB Error: %w", err)
```

Comment:
We should do the check in `runtime/drivers/duckdb` since the DB file blocking is a DuckDB-specific problem (many other databases support multiple processes). 

Also, we should only wrap the error if it's the locked DB error. Something like:

```go
if !strings.Contains(err.Error(), "whatever error duckdb returns when two processes compete") {
// ...
```

Reply:
Updated this change, but error return by driver is very generic i.e `could not open database` not sure it can be the only cause but added this condition